npm/webpack-dev-server/src/makeDefaultWebpackConfig.ts

```diff
@@ -98,7 +99,8 @@ export function makeCypressWebpackConfig (
     devtool: 'inline-source-map',
   } as any
 
-  if (isRunMode) {
+  // if experimentalJustInTimeCompile is configured, we need to watch for file changes as the spec entries are going to be updated per test
```

Comment:
Do we need to watch everything? Or is there a smaller subset of files that we can watch in the JIT case?

Reply:
It's actually doing the later. The watching only applies to files that are in the entry, which is the support file, index file and related files, plus the spec entry. I contemplated narrowing this down to just the specpattern, but we have this weird [hack](https://github.com/cypress-io/cypress/blob/develop/npm/webpack-dev-server/src/CypressCTWebpackPlugin.ts#L122) where we save the index file with no changes to prompt the dev server to trigger a refresh. I wonder if we can just do the index html file and we get the same effect? Ill try it

Reply:
@ryanthemanuel So the watch options for webpack are a bit complicated, but I figure a quick win here is to at least ignore the `node_modules` directory. Vite ignores this by [default](https://vitejs.dev/config/server-options#server-watch). updated watch options in 77b5dee6da